File: ztw-5-spring-swagger/src/main/java/com/github/artsiomshshshsk/bookstore/book/BookService.java
package com.github.artsiomshshshsk.bookstore.book;

import com.github.artsiomshshshsk.bookstore.book.BookController.BookCreateRequest;

import java.util.List;

public interface BookService {
    List<Book> getBooks();

    Book getBookById(int id);

    Book addBook(BookCreateRequest bookCreateRequest);

    Book updateBook(int id, BookController.BookUpdateRequest bookCreateRequest);

    void deleteBook(int id);
}

File: ztw-6-vue/src/api/bookService.js
import axios from 'axios';

const API_URL = '/api/v1/books';

const getBooks = async () => {
    return axios.get(API_URL);
};

const getBookById = async (id) => {
    return axios.get(`${API_URL}/${id}`);
};

const addBook = async (bookData) => {
    return axios.post(API_URL, bookData);
};

const updateBook = async (id, bookData) => {
    return axios.put(`${API_URL}/${id}`, bookData);
};

const deleteBook = async (id) => {
    return axios.delete(`${API_URL}/${id}`);
};

export default {
    getBooks,
    getBookById,
    addBook,
    updateBook,
    deleteBook
};
